packages/grid/_modules_/grid/GridComponent.tsx

```diff
@@ -54,6 +54,26 @@ import { visibleGridRowCountSelector } from './hooks/features/filter/gridFilterS
 
 export const GridComponent = React.forwardRef<HTMLDivElement, GridComponentProps>(
   function GridComponent(props, ref) {
+    const {
+      apiRef: apiRefProp,
+      autoHeight,
+      className,
+      columns,
+      components: componentsProp,
+      componentsProps,
+      getRowId,
+      licenseStatus,
+      loading,
+      nonce,
+      rows,
+      state,
+    } = props;
```

Comment:
Not sure it is worth destructuring the props in this component as the goal is to keep it as short as possible and just have a list of hooks together. 
The idea is to easily recompose it in other subsets, so the more verbose the least obvious it becomes.
Can we have a hook to freeze rows, so we keep this file clean.
useGridFreezeRows or something like that.

Reply:
+1 for moving the freeze into the existing `useGridRows` 👍

Reply:
Well why not treat it seperately. So it could be recomposed.

Reply:
I added it separately in `useGridFreezeRows` as `rows` is used for sorting and filtering hooks as well -- `useGridSorting`, `useGridFilter`. I was thinking about hooks order. So better separate. Now it looks less verbose. Also removed de-structuring of props.

Reply:
I was proposing useGridRows because we likely want to freeze the rows provided with the apiRef. So we could keep the related logic in the same place.

Reply:
I prefer separate as well, as one could decide to use a library like immer or immutable js and this might conflict with it so if it is embedded in useGridRows it would be impossible to just remove this part.
Also I would argue that it is a separate concern, just like useGridEditRows...

Reply:
Ok. The weird part for me is that `useGridEditRows` implements a feature. On the other hand, `useGridFreezeRows` is only here to avoid a foot gun for the developers, to accommodate the implementation tradeoffs of `useGridRows`, relying on immutability. It seems that `useGridFreezeRows` should always come with `useGridRows` (no valid use case without).

Reply:
Well `useGridRows` is so core that it will be difficult to do anything without it.